frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -262,6 +234,86 @@ function DataFrame({
   const { columns, sortColumn, getOriginalIndex, getCellContent } =
     useColumnSort(originalNumRows, originalColumns, getOriginalCellContent)
 
+  const applySelections = React.useCallback(
+    // Use debounce to prevent rapid updates to the widget state.
+    debounce(DEBOUNCE_TIME_MS, (newSelection: GridSelection) => {
+      // If we want to support selections also with in the editable mode,
+      // we would need to integrate the `applyEdits` and `applySelections` functions
+      // into a single function that updates the widget state with both the editing
+      // state and the selection state.
+
+      const selectionState = {
```

Comment:
should we give this a type, like we do with `DataFrameState` on the Python-side?

Reply:
Good point, I added an interface for this